plugins/woocommerce/src/Admin/API/OnboardingPlugins.php

```diff
@@ -236,19 +236,22 @@ public function get_jetpack_authorization_url( WP_REST_Request $request ) {
 		}
 
 		$redirect_url = $request->get_param( 'redirect_url' );
-		$calypso_env  = defined( 'WOOCOMMERCE_CALYPSO_ENVIRONMENT' ) && in_array( WOOCOMMERCE_CALYPSO_ENVIRONMENT, [ 'development', 'wpcalypso', 'horizon', 'stage' ], true ) ? WOOCOMMERCE_CALYPSO_ENVIRONMENT : 'production';
+		$calypso_env  = defined( 'WOOCOMMERCE_CALYPSO_ENVIRONMENT' ) && in_array( WOOCOMMERCE_CALYPSO_ENVIRONMENT, array( 'development', 'wpcalypso', 'horizon', 'stage' ), true ) ? WOOCOMMERCE_CALYPSO_ENVIRONMENT : 'production';
 
-		return [
+		$authorization_url = $manager->get_authorization_url( null, $redirect_url );
+		$authorization_url = add_query_arg( 'locale', explode( '_', get_locale() )[0], $authorization_url );
```

Comment:
There are a few instances where this doesn't work, most notably zh_CN, zh_TW - this is because the param becomes "zh" which is not defined in https://github.com/Automattic/wp-calypso/blob/a93ea780f7ef1df1d3e761af0f79608a1d9ffd89/packages/languages/src/languages-meta.json#L1474 

This leads to a permanently loading screen 
<img width="1326" alt="image" src="https://github.com/user-attachments/assets/72a1cf67-7274-4d90-98e2-fce299224672">

I'm not sure if there's a PHP equivalent of this `@automattic/languages` library that can be used [like this](https://github.com/Automattic/wp-calypso/blob/a93ea780f7ef1df1d3e761af0f79608a1d9ffd89/client/a8c-for-agencies/sections/partner-directory/lib/available-languages.ts#L3)

I did a bit of wrangling and the full list of locales with this issue is here: https://gist.github.com/rjchow/a64fad5dfc219a8fa6cd69cf0869eba4

Reply:
The only one that is fully broken is zh (and bre) because the rest have parent slugs that still match, e.g en_GB will still match en, fr_CA will match fr. However some people might be confused due to dialect differences - e.g es_MX has significant differences from es_ES.

Reply:
Thank you for looking into it @rjchow 🙇 I'll look into it further. We might need to copy the list and use it in PHP 🤔

Reply:
@rjchow Updated in [Return a full locale with language and region code](https://github.com/woocommerce/woocommerce/pull/51392/commits/b93dc7e0810158f3355710a726a8b191ca1ca383)